bcda/main.go

```diff
@@ -439,19 +458,73 @@ func removeExpired(hrThreshold int) {
 
 			id := int(j.ID)
 			jobDir := fmt.Sprintf("%s/%d", os.Getenv("FHIR_PAYLOAD_DIR"), id)
-			expDir = fmt.Sprintf("%s/%d", os.Getenv("FHIR_EXPIRED_DIR"), id)
+			expDir = fmt.Sprintf("%s/%d", os.Getenv("FHIR_ARCHIVE_DIR"), id)
 
 			err = os.Rename(jobDir, expDir)
 			if err != nil {
 				log.Error(err)
 				continue
 			}
 
-			j.Status = "Expired"
+			j.Status = "Archived"
 			err = db.Save(j).Error
 			if err != nil {
 				log.Error(err)
 			}
 		}
 	}
 }
+
+func cleanupArchive(hrThreshold int) error {
+	db := database.GetGORMDbConnection()
+	defer db.Close()
+
+	expDir := os.Getenv("FHIR_ARCHIVE_DIR")
+	if _, err := os.Stat(expDir); os.IsNotExist(err) {
+		// nothing to do if no base directory exists.
+		return nil
+	}
+
+	maxDate := time.Now().Add(-(time.Hour * time.Duration(hrThreshold)))
+
+	var jobs []models.Job
+	err := db.Find(&jobs, "status = ? AND updated_at <= ?", "Archived", maxDate).Error
+	if err != nil {
+		return err
+	}
+
+	if len(jobs) == 0 {
+		log.Info("No archived job files to clean")
+		return nil
+	}
+
+	for _, job := range jobs {
+		t := job.UpdatedAt
+		elapsed := time.Since(t).Hours()
+		if int(elapsed) >= hrThreshold {
+
+			id := int(job.ID)
+			jobArchiveDir := fmt.Sprintf("%s/%d", os.Getenv("FHIR_ARCHIVE_DIR"), id)
+
+			err = os.RemoveAll(jobArchiveDir)
+			if err != nil {
+				log.Error("Unable to remove %s because %s", jobArchiveDir, err)
+				continue
+			}
+
+			job.Status = "Expired"
+			err = db.Save(job).Error
+			if err != nil {
+				return err
+			}
+
+			log.WithFields(log.Fields{
+				"job_began":     job.CreatedAt,
```

Comment:
Also not sure what the exact logging info we want to capture here, but this seems okay to me. Might wanna get additional eyes on this part before merging.

Reply:
I ran this by Ryan yesterday. I think logging is something we should expect to evolve atm, and it's probably better to get something in the logs to work with.

Reply:
agreed, we can always evolve later